src/Drawer/Drawer.js

```diff
@@ -92,34 +87,244 @@ class Drawer extends React.Component {
     firstMount: true,
   };
 
-  componentWillReceiveProps() {
+  componentDidMount() {
+    if (this.props.variant === 'temporary') {
+      this.enableSwipeHandling();
+    }
+    this.mounted = true;
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
This React API is on it's way out. Let's use `getDerivedStateFromProps` instead.